tests/util_test/util.hpp

```diff
@@ -313,8 +319,22 @@ inline auto own_balance(std::shared_ptr<FullNode> const& node) {
   return node->getFinalChain()->getBalance(node->getAddress()).first;
 }
 
-inline auto own_effective_genesis_bal(FullNodeConfig const& cfg) {
-  return cfg.chain.final_chain.state.effective_genesis_balance(dev::toAddress(dev::Secret(cfg.node_secret)));
+inline state_api::BalanceMap effective_genesis_balances(const state_api::Config& cfg) {
+  if (!cfg.dpos) {
+    return cfg.genesis_balances;
+  }
+
+  state_api::BalanceMap effective_balances = cfg.genesis_balances;
+  for (const auto& validator_info : cfg.dpos->initial_validators) {
+    for (const auto& [delegator, amount] : validator_info.delegations) {
+      effective_balances[delegator] -= amount;
```

Comment:
so what happens if validator himself is not among his delegators ? Is this scenario somehow covered ? Because in dpos contract we require self.delegation when registering new validator

Reply:
There should be no issue if we will not self-delegate. We are skipping this checks in initial delegation creation